src/actions/assign.ts

```diff
@@ -0,0 +1,116 @@
+import { Context, Octokit } from "probot"
+import { different } from "../utils"
+
+async function assign(context: Context) {
+  const PRnumber = context.payload.number
+  const owner = context.payload.pull_request.head.user.login
+  const repo = context.payload.pull_request.head.repo.name
+
+  context.log.info("PR number:", PRnumber, ", owner:", owner, ", repo:", repo)
+
+  const oldReviewers: string[] = context.payload.pull_request.requested_reviewers.map(
+    (reviewer: { login: string }) => reviewer.login
+  )
+
+  const newReviewers = await reviewersOfPR(context, PRnumber, owner, repo)
+
+  // Check if old and new reviewers differ. Otherwise we can skip an API call
+  if (different(oldReviewers, newReviewers)) {
+    // Log reviewers replacement
+    context.log.info("Old reviewers:", oldReviewers)
+    context.log.info("New reviewers:", newReviewers)
+
+    const params = context.repo({ pull_number: PRnumber, reviewers: newReviewers })
+    const response = await context.github.pulls.createReviewRequest(params)
+    context.log.info("Status:", response.headers.status)
+  } else {
+    context.log.info("No reviewers changed")
+  }
+}
+
+interface CodeOwner {
+  path: string
+  username: string
+}
+
+function parseModifiedFiles(filesResponse: Octokit.PullsListFilesResponse): string[] {
+  const modifiedFiles: string[] = []
+
+  for (const file of filesResponse) {
+    modifiedFiles.push(file.filename)
+  }
+
+  return modifiedFiles
+}
+
+function parseCodeOwners(codeownersFile: string): CodeOwner[] {
+  const codeOwners: CodeOwner[] = []
+  for (const line of codeownersFile.split("\n")) {
+    if (line.length === 0 || line.startsWith("#")) {
+      continue
+    }
+
+    let [path, owner] = line.split(" ")
+
+    // Ignore first line and empty newline at the end of the file
+    if (!owner?.startsWith("@")) {
+      continue
+    }
+
+    owner = owner.replace("@", "")
```

Comment:
Let's replace this with `owner.slice(1)` which just removes the first character ([seems to be faster](https://jsperf.com/owner-slice-vs-replace), but more importantly it is cleaner).